conn.go

```diff
@@ -394,6 +393,44 @@ func (c *Conn) RemoveBPF() error {
 	return s.RemoveBPF()
 }
 
+// A deadlineSetter is a Socket that supports setting deadlines.
+type deadlineSetter interface {
+	Socket
+	SetDeadline(time.Time) error
+	SetReadDeadline(time.Time) error
+	SetWriteDeadline(time.Time) error
+}
+
+// SetDeadline sets the read and write deadlines associated with the connection.
```

Comment:
I think it's worth adding a note to each of these functions about their Go version compatibility:

```go
// SetDeadline sets the read and write deadlines associated with the connection.
//
// Deadline functionality is only supported on Go 1.12+. Calling this function on older versions
// of Go will result in an error.
```